File: 3111. Minimum Rectangles to Cover Points/Solution.cpp
#include <ranges>
#include <vector>
using namespace std;

class Solution {
public:
    int minRectanglesToCoverPoints(vector<vector<int>>& points, int w)
    {
        ranges::sort(points, [](const vector<int>& left, const vector<int>& right) { return left[0] < right[0]; });
        int res = 1, beg = points[0][0];
        for (const vector<int>& p : points) {
            if (p[0] - beg > w) {
                res++;
                beg = p[0];
            }
        }
        return res;
    }
};

File: 3111. Minimum Rectangles to Cover Points/Solution.py
from typing import List


class Solution:
    def minRectanglesToCoverPoints(self, points: List[List[int]], w: int) -> int:
        points.sort(key=lambda x: x[0])
        res, beg = 1, points[0][0]
        for x, _ in points:
            if x - beg > w:
                res += 1
                beg = x
        return res
